### PR TITLE
Add comprehensive test coverage for all mint database functions

### LINKED ISSUE #632: Create database integrations tests
We should add database specific tests to our integration tests that access the dbs directly and confirm db layers act as expected ie cannot remove spent proofs. 

cc @crodas believe you mentioned something along these lines. 

related: https://github.com/cashubtc/cdk/pull/630#issuecomment-2702097400

### PR DESCRIPTION
**Pull request by crodas**
### Description

Fixes #632

Add 19 new tests achieving 100% coverage of mint database traits:

Saga Database (12 tests):
- add_and_get_saga, add_duplicate_saga, update_saga_state
- delete_saga, get_incomplete_swap_sagas, get_incomplete_melt_sagas
- get_nonexistent_saga, update_nonexistent_saga, delete_nonexistent_saga
- saga_with_quote_id, saga_transaction_rollback, multiple_sagas_different_states

QuotesTransaction & SignaturesTransaction (7 tests):
- increment_mint_quote_amount_paid, increment_mint_quote_amount_issued
- get_mint_quote_in_transaction, get_melt_quote_in_transaction
- get_mint_quote_by_request_in_transaction
- get_mint_quote_by_request_lookup_id_in_transaction
- get_blind_signatures_in_transaction

Test suite expanded from 53 to 72 tests, all passing for SQLite backend. Tests verify transaction isolation, rollback behavior, state management, and proper handling of edge cases across all database implementations.

#### Follow up

- Once #1204 gets merged, add tests to wallet database traits as well.

-----

### Notes to the reviewers

-----

#### FIXED

----

### Checklist

* [x] I followed the [code style guidelines](https://github.com/cashubtc/cdk/blob/main/CODE_STYLE.md)
* [x] I ran `just final-check` before committing

### CODE REVIEW

File: crates/cdk-common/src/database/mint/test/keys.rs [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+
+    // Add the same keyset info again - this may succeed (idempotent) or fail
+    // Both behaviors are acceptable
+    let mut tx = KeysDatabase::begin_transaction(&db).await.unwrap();
+    let result = tx.add_keyset_info(keyset_info).await;
+    if result.is_ok() {
+        tx.commit().await.unwrap();
+    } else {
+        tx.rollback().await.unwrap();
+    }
```
Comment (thesimplekid) [inline crates/cdk-common/src/database/mint/test/keys.rs]: Shouldn't we pick one?

File: crates/cdk-common/src/database/mint/test/mint.rs [hunk 2]
```diff
@@ -600,3 +600,675 @@ where
...
+    assert_eq!(new_total, 500.into());
+    tx.commit().await.unwrap();
+
+    // Verify final state
+    let retrieved = db.get_mint_quote(&mint_quote.id).await.unwrap().unwrap();
+    assert_eq!(retrieved.amount_paid(), 500.into());
+}
+
+/// Test incrementing mint quote amount issued
+pub async fn increment_mint_quote_amount_issued<DB>(db: DB)
```
Comment (thesimplekid) [inline crates/cdk-common/src/database/mint/test/mint.rs:1032]: Think we should add test we get error if w try to increment with the same payment id

File: crates/cdk-common/src/database/mint/test/proofs.rs [hunk 3]
```diff
@@ -162,3 +162,457 @@ where
         "Duplicate entry"
     );
 }
+
+/// Test updating proofs states
+pub async fn update_proofs_states<DB>(db: DB)
```
Comment (thesimplekid) [inline crates/cdk-common/src/database/mint/test/proofs.rs:167]: Do we still check for valid state transitions in db or we expect that in application

Comment (crodas) [reply crates/cdk-common/src/database/mint/test/proofs.rs:167]: Yes, I will create a follow-up to move the check to the mint and not the storage layer.

Comment (thesimplekid) [reply crates/cdk-common/src/database/mint/test/proofs.rs:167]: I think we should still have a check in the db that we do not delete `SPENT` proofs.

File: crates/cdk-common/src/database/mint/test/mint.rs (hunk 2 above)
Comment (crodas) [reply crates/cdk-common/src/database/mint/test/mint.rs:1032]: done
